getting_started/workflow/best_practices/what_are_godot_classes.rst

```diff
@@ -1,127 +1,66 @@
+derived types using a script.
 
-Let's see a simple example of creating a single ``Node`` as a child. The code
-below creates a new ``Node``, changes its name, assigns a script to it, sets its
-future parent as its owner so it gets saved to disk along with it, and finally
-adds it as a child of the ``Main`` node:
+These scripts are not technically classes. Instead, they are resources that tell the engine a
+sequence of initializations to perform on one of the engine's built-in classes.
 
-.. tabs::
-  .. code-tab:: gdscript GDScript
+Godot's internal classes have methods that register a class's data with a :ref:`ClassDB
+<class_ClassDB>`. This database provides runtime access to class information. ``ClassDB`` contains
+information about classes like:
 
-    # Main.gd
-    extends Node
+- Properties.
+- Methods.
+- Constants.
+- Signals.
 
-    func _init():
-        var child = Node.new()
-        child.name = "Child"
-        child.script = preload("Child.gd")
-        child.owner = self
-        add_child(child)
+This ``ClassDB`` is what objects check against when performing an operation like accessing a
+property or calling a method. It checks the database's records and the object's base types' records
+to see if the object supports the operation.
 
-  .. code-tab:: csharp
+Attaching a :ref:`Script <class_Script>` to your object extends the methods, properties, and signals
+available from the ``ClassDB``.
 
-    using System;
-    using Godot;
+.. note::
 
-    namespace ExampleProject
-    {
-        public class Main : Resource
-        {
-            public Node Child { get; set; }
+    Objects check their attached script before the database, which is why scripts can override
+    built-in methods. If a script defines a ``_get_property_list()`` method, Godot merges that data
+    with the list of properties the object fetches from the ClassDB. In case of a shared name, Godot
+    replaces the existing property. The same is true for other declarative code.
 
-            public Main()
-            {
-                Child = new Node();
-                Child.Name = "Child";
-                Child.Script = (Script)ResourceLoader.Load("child.gd");
-                Child.Owner = this;
-                AddChild(Child);
-            }
-        }
-    }
+    Even scripts that don't use the ``extends`` keyword implicitly inherit from the engine's base
+    :ref:`Reference <class_Reference>` class. This implicit type allows the object to defer to the
+    script's content where the engine logic deems appropriate.
 
-Script code like this is much slower than engine-side C++ code. Each change
-makes a separate call to the scripting API which leads to many "look-ups" on the
-back-end to find the logic to execute.
+    As a result, you can instance scripts without the ``extends`` keyword from code, but you cannot
+    attach them to a :ref:`Node <class_Node>`.
```

Comment:
```rst
As a result, you can instantiate scripts without the ``extends`` keyword from code.
Since they extend ``Reference`` though, you cannot attach them to a :ref:`Node <class_Node>`.
```

Reason: more clarity.